### PR TITLE
Fix alignment between line and code columns.

### PR DESCRIPTION
**Pull request by pelmers**
We stop stripping empty lines in multi-line regions,
especially prevent in Python docstrings.

Example of the issue: https://dxr.allizom.org/mozilla-central/source/python/mozbuild/mozbuild/frontend/context.py#1218
Here the highlighted line number is about a hundred lines below the highlighted line of source code.

### CODE REVIEW

**Comment by erikrose** [comment]
I get some failing tests:

```
FAIL: test_lines:IntegrationTests.test_split_anchor_across_lines
  vi +272 tests/test_lines.py  # test_split_anchor_across_lines
    [u'<a data-menu="{}">this</a>', u'<a data-menu="{}">that</a>'])
AssertionError: [Markup(u'<a data-menu="{}">this\n</a>'), Markup(u'<a data-menu="{}">that</a>')] != [u'<a data-menu="{}">this</a>', u'<a data-menu="{}">that</a>']

FAIL: test_vcs_git:GitTests.test_pygmentize
  vi +42 tests/test_vcs_git/test_vcs_git.py  # test_pygmentize
    ok_('<span class="c">// Hello World Example</span>' in response.data)
AssertionError: None
```

File: tests/test_lines.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def test_split_anchor_avoidance(self):
     def test_split_anchor_across_lines(self):
         """Support unavoidable splits of an anchor across lines."""
         eq_(text_to_html_lines('this\nthat', refs=[(0, 9, Ref({}))]),
-            [u'<a data-menu="{}">this</a>', u'<a data-menu="{}">that</a>'])
+            [u'<a data-menu="{}">this\n</a>', u'<a data-menu="{}">that</a>'])
```
Comment (pelmers) [inline tests/test_lines.py]: I wonder if having this will cause any display issues on the page.

Comment (erikrose) [inline tests/test_lines.py]: How would it? It's the \n within region tags, anyway, that fixes the bug.

Comment (erikrose) [inline tests/test_lines.py]: Let's add a comment explaining that preserving the \n is necessary to correctly display line breaks within regions or refs. Then, r+.

**Pull request merged**